controllers/operands/tto.go

```diff
@@ -0,0 +1,117 @@
+package operands
+
+import (
+	"errors"
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ttov1alpha1 "github.com/kubevirt/tekton-tasks-operator/api/v1alpha1"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+type ttoHandler struct {
+	genericOperand
+}
+
+func newTtoHandler(Client client.Client, Scheme *runtime.Scheme) *ttoHandler {
+	return &ttoHandler{
+		genericOperand: genericOperand{
+			Client:                 Client,
+			Scheme:                 Scheme,
+			crType:                 "TektonTasks",
+			removeExistingOwner:    false,
+			setControllerReference: false,
+			hooks:                  &ttoHooks{},
+		},
+	}
+}
+
+type ttoHooks struct {
+	cache *ttov1alpha1.TektonTasks
+}
+
+func (h *ttoHooks) getFullCr(hc *hcov1beta1.HyperConverged) (client.Object, error) {
+	if h.cache == nil {
+		tto, err := NewTTO(hc)
+		if err != nil {
+			return nil, err
+		}
+		h.cache = tto
+	}
+	return h.cache, nil
+}
+
+func (h ttoHooks) getEmptyCr() client.Object { return &ttov1alpha1.TektonTasks{} }
+func (h ttoHooks) getConditions(cr runtime.Object) []metav1.Condition {
+	return osConditionsToK8s(cr.(*ttov1alpha1.TektonTasks).Status.Conditions)
+}
+func (h ttoHooks) checkComponentVersion(cr runtime.Object) bool {
+	found := cr.(*ttov1alpha1.TektonTasks)
+	return checkComponentVersion(hcoutil.TtoVersionEnvV, found.Status.ObservedVersion)
+}
+func (h ttoHooks) getObjectMeta(cr runtime.Object) *metav1.ObjectMeta {
+	return &cr.(*ttov1alpha1.TektonTasks).ObjectMeta
+}
+func (h *ttoHooks) reset() {
+	h.cache = nil
+}
+
+func (h *ttoHooks) updateCr(req *common.HcoRequest, client client.Client, exists runtime.Object, required runtime.Object) (bool, bool, error) {
+	tto, ok1 := required.(*ttov1alpha1.TektonTasks)
+	found, ok2 := exists.(*ttov1alpha1.TektonTasks)
+	if !ok1 || !ok2 {
+		return false, false, errors.New("can't convert to TTO")
+	}
+	if !reflect.DeepEqual(found.Spec, tto.Spec) ||
+		!reflect.DeepEqual(found.Labels, tto.Labels) {
+		if req.HCOTriggered {
+			req.Logger.Info("Updating existing TTO's Spec to new opinionated values")
+		} else {
+			req.Logger.Info("Reconciling an externally updated TTO's Spec to its opinionated values")
+		}
+		util.DeepCopyLabels(&tto.ObjectMeta, &found.ObjectMeta)
+		tto.Spec.DeepCopyInto(&found.Spec)
+		err := client.Update(req.Ctx, found)
+		if err != nil {
+			return false, false, err
+		}
+		return true, !req.HCOTriggered, nil
+	}
+	return false, false, nil
+}
+
+func NewTTO(hc *hcov1beta1.HyperConverged, opts ...string) (*ttov1alpha1.TektonTasks, error) {
```

Comment:
Is there any error case here? please consider to drop the error (this function does not implement any interface)